lib/directory-monitor-poll.c

```diff
@@ -85,16 +106,37 @@ _rescan_directory(DirectoryMonitorPoll *self)
 {
   GError *error = NULL;
   GDir *directory = g_dir_open(self->super.real_path, 0, &error);
+  gchar *full_path = NULL;
+  GStatBuf st;
+  CollectionComparatorEntry *entry;
+
   collection_comparator_start(self->comparator);
   if (directory)
     {
       const gchar *filename;
       while((filename = g_dir_read_name(directory)))
         {
           collection_comparator_add_value(self->comparator, filename);
+
+          full_path = build_filename(self->super.real_path, filename);
+          entry = g_hash_table_lookup(self->comparator->original_map, filename);
+
+          if (g_stat(full_path, &st))
+            perror(full_path);
+          else if (self->initial_rescan)
+            entry->mtime = st.st_mtime;
```

Comment:
If the current file is not in the map, this line is a null pointer dereference.

Reply:
Could there be a case where it not going to be in the map after ``collection_comparator_add_value(self->comparator, filename);`` has just been called?

Reply:
Well at this level the `collection_comparator_add_value` is black box, but I would say move these things inside, and probably this won't be an issue.

Reply:
I judged based on the fact that ``collection_comparator_add_value`` will either add an entry if not exists or update the entry flag if it already exists, thus, there will always be an entry. But I see your perspective.